lms/static/scripts/frontend_apps/errors.js

```diff
@@ -8,8 +8,26 @@ export class PickerCanceledError extends Error {
 }
 
 /**
- * Error returned when an API call fails with a 4xx or 5xx response and
- * JSON body.
+ * @typedef {import('./config').ConfigErrorBase} ConfigErrorBase
+ */
+
+/**
+ * Error returned when error data is provided in application configuration JSON
+ * and application is in an error mode ('error-dialog' or 'oauth2-redirect-error')
+ */
+export class AppConfigError extends Error {
+  /**
+   * @param {ConfigErrorBase} data
+   */
+  constructor(data) {
+    super();
```

Comment:
Subclasses of error should always set a sensible message via `super(<message>)`, since that gets used if the error is ever handled by something which understands `Error` but not the specific subclass. For example, browser console logs.